dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/S3ClientHelper.java

```diff
@@ -105,4 +106,12 @@ public static String getFileName(String key) {
     public static String[] splitKey(String key) {
         return key.split(KEY_DELIMITER);
     }
+
+    /**
+     * Creates a file name using the current time in milliseconds since epoch appended with '.json'
+     * @return
+     */
+    public static String createFileName() {
+        return Instant.now().toEpochMilli() + ".json";
```

Comment:
This will work 99.9999999999999-100% of the time, so no need to change. But, just curious, what would happen if 2 requests were processed at the same millisecond? Overwriting or an s3 error?

Reply:
> This will work 99.9999999999999-100% of the time, so no need to change. But, just curious, what would happen if 2 requests were processed at the same millisecond? Overwriting or an s3 error?

For an instant, the old and new S3 objects are superposed, pricking a hole in the fabric of space and time that, expanding ever-more-rapidly outwards, consumes the universe.

But seriously, yeah, this worried me about the last PR and still does, even though it will never happen. :)

Reply:
You could add on the nanoseconds :wink: 
https://docs.oracle.com/javase/8/docs/api/java/time/Instant.html#getNano--

Reply:
> what would happen if 2 requests were processed at the same millisecond? Overwriting or an s3 error?

Since the metrics bucket doesn't have versioning, I believe the second request will overwrite the first